### PR TITLE
Adds initial support to connect to token backend

### LINKED ISSUE #885: Implement tokens screen
Need to implement token management screen with a link in the left nav of the
application.

Table should have a type column indicating whether it is an OAuth or SA type
(this should be made available after the token has been created by the controller
annotating this information on the `MigToken` object inside `status`). It will
list the associated cluster, as well as the expiration (also set by the controller
in the status field).

The tokens each have a "Status" that indicates if its Expired, Expiring soon,
or ok.

Expiring soon threshold should be easy to change, though probably doesn't require a
configmap value. Maybe 2 hours to start with.

### LINKED ISSUE #891: Add MigToken support to k8s client
Need to add MigToken support to the CAM UI's k8s client so we have API access
similar to the rest of the objects.

### PR DESCRIPTION
**Pull request by eriknelson**
This PR is another WIP PR adding incremental non-admin feature support; it includes:
* Adds MigToken support to the client
* Adds token polling
* Hooks up the Tokens table to the redux tree and removes fake tokens
* Adds "Add Token" support for SAs
* Some amount of refactoring where it made sense (ex: moving the token form values into the token types, since it is used in at least 2 places other than the form component itself)

It does **not** include:
* Token edit/delete
* OAuth token support (will follow up with this)

I also discovered #917 while implementing this but refrained from fixing it and adding scope creep. I'd like to follow up this PR with a distinct PR to repair this issue as well.

Closes #891
Closes #885

### CODE REVIEW

File: src/client/resources/conversions.ts [hunk 1]
```diff
@@ -682,6 +682,50 @@ export function createMigHook(migHook: any, namespace: string) {
...
+  namespace: string,
+  migTokenSecretName: string,
+  migTokenSecretNamespace: string,
+  migClusterName: string
+) {
+  return {
+    apiVersion: 'migration.openshift.io/v1alpha1',
+    kind: 'MigToken',
+    metadata: {
+      name,
```
Comment (ibolton336) [inline src/client/resources/conversions.ts:713]: should we use the generateName field for the token name too?

File: src/app/token/duck/sagas.ts [hunk 2]
```diff
@@ -1,2 +1,156 @@
...
+
+function* watchAddTokenRequest() {
+  yield takeLatest(TokenActionTypes.ADD_TOKEN_REQUEST, addTokenRequest);
+}
+
+export default {
+  // NATODO: Implement and/or remove unecessary copies
+  // watchRemoveClusterRequest,
+  watchAddTokenRequest,
+  // watchUpdateClusterRequest,
```
Comment (ibolton336) [inline src/app/token/duck/sagas.ts:153]: cleanup

File: src/app/token/duck/actions.ts [hunk 3]
```diff
@@ -36,4 +53,33 @@ const migTokenFetchFailure = () => ({
...
+  params,
+});
+
+const stopTokenPolling = () => ({
+  type: TokenActionTypes.TOKEN_POLL_STOP,
+});
+
+// NATODO: Implement and/or remove unecessary copies
+export const TokenActions = {
+  // removeClusterRequest,
```
Comment (ibolton336) [inline src/app/token/duck/actions.ts:67]: cleanup

File: src/client/resources/conversions.ts (hunk 1 above)
Comment (eriknelson) [reply src/client/resources/conversions.ts:713]: Don't think so. It's a primary Mig object like a MigCluster or MigStorage object and the user provides a name. We've discussed maybe decoupling the name the user provides from the actual name that's used on the object so it doesn't have to abide by kubernetes naming requirements as a primary key, but I think that's distinctly separate from this.

**Comment by vconzola** [comment]
@ibolton336 That's correct. We should use an empty state component inside of table as shown here: https://www.patternfly.org/v4/design-guidelines/usage-and-behavior/empty-state

**Comment by eriknelson** [comment]
Added empty state for tokens:
image

**Comment by mturley** [comment]
Hey, sorry I didn't look at this sooner. I had originally left out the empty state for tokens because Vince told me to, since we assumed the user would always have at least the one token they logged in with.

**Comment by vconzola** [comment]
Sure @mturley, blame it on the designer who doesn't know anything.  😉  @eriknelson didn't you tell me that users will always have at least one token? That's what I have in my notes because "user can't delete the host cluster token".

**Comment by eriknelson** [comment]
Yeah I forgot about that, I don't expect the host token to ever be invalid or to be removable. It's exposed here because we don't have the auto creation of the logged in token implemented yet.

**Comment by mturley** [comment]
Haha, sorry Vince. I was pretty confident you were right, if that helps 🙃

**Pull request merged**